src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -140,4 +162,146 @@ public void parseLoyaltyPoints_validValueWithWhitespace_returnsLoyaltyPoints() t
         LoyaltyPoints expectedLoyaltyPoints = new LoyaltyPoints(VALID_LOYALTY_POINTS_INT);
         assertEquals(expectedLoyaltyPoints, ParserUtil.parseLoyaltyPoints(loyaltyPointsWithWhitespace));
     }
+
+    @Test
+    public void parseIngredientName_invalidNonAlphabetValue_throwsExcepion() throws Exception {
+        thrown.expect(ParseException.class);
+        ParserUtil.parseIngredientName(INVALID_INGREDIENTNAME_INTEGER);
+    }
+
+    @Test
+    public void parseIngredientName_invalidSymbolsValue_throwsExcepion() throws Exception {
+        thrown.expect(ParseException.class);
+        ParserUtil.parseIngredientName(INVALID_INGREDIENTNAME_SYMBOLS);
+    }
+
+    @Test
+    public void parseIngredientName_null_throwsException() throws Exception {
+        Assert.assertThrows(NullPointerException.class, () -> ParserUtil.parseIngredientName((String) null));
+    }
+
+
+    @Test
+    public void parseIngredientName_validValue_returnsIngredientName() throws Exception {
+        String ingredientNameWithWhiteSpace = WHITESPACE + VALID_INGREDIENTNAME + WHITESPACE;
+        IngredientName expectedIngredientName = new IngredientName(VALID_INGREDIENTNAME);
+
+        //with white space
+        assertEquals(expectedIngredientName, ParserUtil.parseIngredientName((ingredientNameWithWhiteSpace)));
+
+        //without white space
+        assertEquals(expectedIngredientName, ParserUtil.parseIngredientName((VALID_INGREDIENTNAME)));
+    }
+
+    @Test
+    public void parseIngredientQuantity_invalidNegativeValue_throwsExcepion() throws Exception {
+        thrown.expect(ParseException.class);
+        ParserUtil.parseIngredientQuantity(Integer.toString(INVALID_INGREDIENTQUANTITY_NEGATIVE));
+    }
+
+    @Test
+    public void parseIngredientQuantity_invalidOutOfRange_throwsExcepion() throws Exception {
+        thrown.expect(ParseException.class);
+        ParserUtil.parseIngredientQuantity(Long.toString(Integer.MAX_VALUE + 1));
```

Comment:
Integer.MAX_VALUE + 1 overflows to Integer.MIN_VALUE. Should convert Integer to Long before adding 1